test/ssl_test_ctx.c

```diff
@@ -95,6 +95,7 @@ static const test_enum ssl_test_results[] = {
     {"ServerFail", SSL_TEST_SERVER_FAIL},
     {"ClientFail", SSL_TEST_CLIENT_FAIL},
     {"InternalError", SSL_TEST_INTERNAL_ERROR},
+    {"TimeoutError", SSL_TEST_TIMEOUT_ERROR},
```

Comment:
I think a better fix is below. This means you can remove all of the TimeoutError stuff:

````
Author:     Matt Caswell <matt@openssl.org>
AuthorDate: Fri May 12 13:45:22 2017 +0100
Commit:     Matt Caswell <matt@openssl.org>
CommitDate: Fri May 12 13:45:22 2017 +0100

    Always flush the BIO when we send any alert
    
    At the moment we flush the write BIO if we send a fatal alert, but not a
    warning one. This can mean the warning is never sent if we never do another
    write and subsequently flush the BIO. Instead we should just always flush
    after writing an alert.

diff --git a/ssl/s3_msg.c b/ssl/s3_msg.c
index 7af2f99e05..1cd3941a27 100644
--- a/ssl/s3_msg.c
+++ b/ssl/s3_msg.c
@@ -105,12 +105,10 @@ int ssl3_dispatch_alert(SSL *s)
         s->s3->alert_dispatch = 1;
     } else {
         /*
-         * Alert sent to BIO.  If it is important, flush it now. If the
-         * message does not get sent due to non-blocking IO, we will not
-         * worry too much.
+         * Alert sent to BIO - now flush. If the message does not get sent due
+         * to non-blocking IO, we will not worry too much.
          */
-        if (s->s3->send_alert[0] == SSL3_AL_FATAL)
-            (void)BIO_flush(s->wbio);
+        (void)BIO_flush(s->wbio);
 
         if (s->msg_callback)
             s->msg_callback(1, s->version, SSL3_RT_ALERT, s->s3->send_alert,
````